_posts/2024-04-24-highlights-otp-27.md

```diff
@@ -0,0 +1,1048 @@
+---
+layout: post
+title: Erlang/OTP 27 Highlights
+tags: erlang otp 27 release
+author: Björn Gustavsson
+---
+
+Erlang/OTP 27 is finally here. This blog post will introduce the new
+features that we are most excited about.
+
+A list of all changes is found in [Erlang/OTP 27 Readme](/patches/OTP-27.0).
+Or, as always, look at the release notes of the application you are interested in.
+For instance:
+[Erlang/OTP 27 - Erts Release Notes - Version 15.0](https://www.erlang.org/doc/apps/erts/notes.html#erts-15.0).
+
+This year's highlights mentioned in this blog post are:
+
+* [Overhauled documentation system](#overhauled-documentation-system)
+* [Triple-Quoted strings](#triple-quoted-strings)
+* [Sigils](#sigils)
+* [No need to enable feature `maybe`](#no-need-to-enable-feature-maybe)
+* [The new `json` module](#the-new-json-module)
+* [Process labels](#process-labels)
+* [New functionality in STDLIB](#new-functionality-in-stdlib)
+* [Multiple trace sessions](#multiple-trace-sessions)
+* [Native coverage support](#native-coverage-support)
+* [Deprecating archives](#deprecating-archives)
+
+# Overhauled documentation system
+
+The Erlang/OTP documentation before Erlang/OTP 27 was authored in
+[XML](https://en.wikipedia.org/wiki/XML), from which the
+[Erl_Docgen](https://www.erlang.org/docs/26/apps/erl_docgen/)
+application could generate HTML web pages, PDFs, or Unix man pages.
+The reason for generating PDFs is that the documentation used to be
+printed as
+[actual paper books](https://erlangforums.com/t/old-printed-otp-documentation-cover/1989/2?u=bjorng).
+The last time the books were printed were for the Erlang/OTP R7 released in 2000.
+
+As an example, here is the XML code for
+[`lists:duplicate/2`](https://www.erlang.org/docs/26/man/lists#duplicate-2)
+from Erlang/OTP 26:
+
+```xml
+    <func>
+      <name name="duplicate" arity="2" since=""/>
+      <fsummary>Make <c>N</c> copies of element.</fsummary>
+      <desc>
+        <p>Returns a list containing <c><anno>N</anno></c> copies of term
+          <c><anno>Elem</anno></c>.</p>
+        <p><em>Example:</em></p>
+        <pre>
+> <input>lists:duplicate(5, xx).</input>
+[xx,xx,xx,xx,xx]</pre>
+      </desc>
+    </func>
+```
+
+The XML code was stored in separate files, not in the source
+code. When building the documentation, the function specs from the
+source code would be combined with the text from the documentation
+file. It was the responsibility of the writer to ensure that variables
+mentioned in the documentation body matched the names in the function
+spec.
+
+One thing never said about Erl_Docgen and the old documentation system
+was that it made writing documentation enjoyable and effortless. That
```

Comment:
Yes, it is a slight attempt to joke, and no, I didn't like writing documentation in XML.

As I read it, no one ever said "it made writing documentation enjoyable and effortless", because no one finds it enjoyable and effortless.

Reply:
FWIW, you got a chuckle out of me for that line...